### DISCUSSION TITLE
CI for stationary probabilities

### DISCUSSION THREAD

**Discussion by FedericoDP** [bmcclintock/momentuHMM#72] [Q&A] [answered]
Dear Brett,

I used the stationary function to calculate the stationary probabilities based on different covariate values in a 3-states HMM. I am now wondering if there is any way to obtain a CI and/or SE for the stationary probabilities obtained? I see that these are calculated and plotted within 'plotStationary', but it would be great to extract the values (similarly to what CIreal does for state transition probabilities). Is there a function that I somehow missed?

Thank you for any insight,
All the best,
Federico

**Comment by bmcclintock** [comment]
Hi Federico,

I might need to add this to my to-do list, as you are correct that there is currently no way for the CI and/or SE to be returned by a momentuHMM function. A hack would be to use:

`debug(momentuHMM:::plotCall)`
`plotStationary(example$m,plotCI=TRUE)`

and manually save the `se` (standard error), `lci` (lower CI), and `uci` (upper CI) for each state and covariate combination while in debug mode.

Cheers,
Brett

**Comment by FedericoDP** [comment]
Hi Brett,
Thank you very much for your reply, the hack does the work, it's very useful!

Cheers,
Federico

**Comment by DEHewitt** [comment]
Hi @bmcclintock, just checking in to see if this did ever get added to your to-do list or if the above workaround is still the best way. Cheers!

**Comment by bmcclintock** [comment, marked as answer]
A `return` argument has been added to the `plotStationary` function. When `plotCI=TRUE` and `return=TRUE`, the stationary distribution estimates, SEs, and CIs are returned as a list with elements for each mixture, covariate, and state combination. This feature is currently only available in the 'develop' branch:

`> remotes::install_github('bmcclintock/momentuHMM@develop)`

but will be included in the next CRAN release (version `>=1.5.3`).
